index.js

```diff
@@ -30,26 +27,26 @@ module.exports = function(opts) {
     }
 
     if (!token && !opts.passthrough) {
-      this.throw(401, 'No authentication token found\n');
+      ctx.throw(401, 'No authentication token found\n');
     }
 
-    secret = (this.state && this.state.secret) ? this.state.secret : opts.secret;
+    secret = (ctx.state && ctx.state.secret) ? ctx.state.secret : opts.secret;
     if (!secret) {
-      this.throw(500, 'Invalid secret\n');
+      ctx.throw(500, 'Invalid secret\n');
     }
 
     try {
-      user = yield JWT.verify(token, secret, opts);
+      user = JWT.verify(token, secret, opts);
```

Comment:
I wrote it as an async call (using bluebird's promisify), but then I recall you guys asked me to "remove promisification" of JWT methods. And so I did. Maybe I misunderstood you.